testsuite/integration-arquillian/tests/other/console/src/main/java/org/keycloak/testsuite/console/page/clients/mappers/CreateClientMappersForm.java

```diff
@@ -117,14 +118,21 @@ public void selectRealmRole(String roleName) {
         }
         
         public void selectClientRole(String clientName, String roleName) {
+            WaitUtils.pause(100);
             if (roleName != null || clientName != null) {
-                clientSelect.selectByVisibleText(clientName);
                 clientAvailable.selectByVisibleText(roleName);
             }
             WaitUtils.pause(1000);
             selectClientRoleButton.click();
             WaitUtils.waitForModalFadeOut();
         }
+
+        public void selectClient(String clientName) {
+            driver.findElement(By.cssSelector("#s2id_clients .select2-choice")).click();
+            String subContainerClass = "#select2-drop:not([style*='display: none'])";
+            driver.findElement(By.cssSelector( subContainerClass + " .select2-input")).sendKeys(clientName);
+            driver.findElements(By.cssSelector(subContainerClass + " .select2-results li.select2-result-selectable")).get(0).click();
```

Comment:
Could you perhaps try to use similar approach to [this](https://github.com/keycloak/keycloak/blob/master/testsuite/integration-arquillian/tests/other/console/src/main/java/org/keycloak/testsuite/console/page/clients/authorization/policy/ClientPolicyForm.java#L89)? The point is not to copy-paste code between Page Objects and also not to put so many ugly selectors directly into the methods.

Reply:
@saibot94 Please, let me know if you're working on this and if I can help you with this somehow. It's actually not really that blocking and I'm willing to approve this as is. It's just something nice to have as it's definitely a reusable peace of code as the Client selector is now used at many places throughout the Admin Console – so it'd make sense to make it a Page Fragment as I mentioned.

Reply:
Hey, you can check out the `ClientSelectModal` class I've added. It hides away the ugly CSS selectors in `ClientScopesEvaluateForm` and `CreateClientMappersForm`.

I think you wanted something like this, the tests run just fine.